src/features/PostState/usePosts.js

```diff
@@ -19,16 +34,17 @@ export const PostsProvider = ({ children }) => {
                 sortDirection: queryParams?.sortDirection,
                 searchTerm: queryParams?.searchTerm,
             })
-            .firstPage((err, records) =>
+            .eachPage((records, fetchNextPage) => {
+                retrieveNextPage.current = fetchNextPage;
                 postsDispatcher({
-                    type: 'FETCH_POSTS_SUCCESSFUL',
+                    type: 'ADD_PAGE_OF_POSTS',
                     payload: {
-                        posts: records.map(records => records.fields),
+                        newPosts: records.map(records => records.fields),
                         sortDirection: queryParams?.sortDirection,
                         searchTerm: queryParams?.searchTerm,
                     },
-                })
-            );
+                });
+            }, endPagination);
```

Comment:
The second argument of `eachPage` takes a callback for what to do when the paginating is done, i.e. we retrieved all pages.